File: leetcode/hot100/codes/Go/杨辉三角.go
func generate(numRows int) [][]int {
	ans := make([][]int, numRows)

	for i := range numRows {
		ans[i] = make([]int, i+1)
		ans[i][0] = 1
		ans[i][i] = 1

		for j := 1; j < i; j++ {
			ans[i][j] = ans[i-1][j-1] + ans[i-1][j]
		}
	}

	return ans
}

File: leetcode/hot100/codes/Python/杨辉三角.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = list()
        for i in range(numRows):
            row = list()
            for j in range(0, i + 1):
                if j == 0 or j == i:
                    row.append(1)
                else:
                    row.append(ans[i - 1][j] + ans[i - 1][j - 1])
            ans.append(row)
        
        return ans